### PR TITLE
[FIX] added API endpoint for im.history.others, as advertised on rocket.chat REST API settings

### PR DESCRIPTION
**Pull request by BlackFenix2**
This pull request adds an API endpoint for im.history.others as advertised on the rocket.chat REST API settings.

this was added to allow latest and oldest datetime parameters to be used for direct messages the user is not a part of.

image

### CODE REVIEW

File: packages/rocketchat-api/server/v1/im.js [hunk 1]
```diff
@@ -177,6 +177,68 @@ RocketChat.API.v1.addRoute(['dm.history', 'im.history'], { authRequired: true },
 	},
 });
 
+RocketChat.API.v1.addRoute(['dm.history.others', 'im.history.others'], { authRequired: true }, {
+	get() {
+		// const findResult = findDirectMessageRoom(this.requestParams(), this.user);
```
Comment (MarcosSpessatto) [inline packages/rocketchat-api/server/v1/im.js]: Can you please remove this commented code?

Comment (BlackFenix2) [reply packages/rocketchat-api/server/v1/im.js]: i just updated my branch and removed the comment.

**Comment by BlackFenix2** [comment]
@MarcosSpessatto Sorry if this is a stupid question, where is the tests for the v1/im.history endpoint?

**Comment by BlackFenix2** [comment]
Sorry, i found the test area, its under https://github.com/RocketChat/Rocket.Chat/blob/5b3ca5da176c3df07ca1d2a70851f936ee9a479b/tests/end-to-end/api/04-direct-message.js

ill add a test for this endpoint.

**Comment by BlackFenix2** [comment]
> @BlackFenix2 can you add some tests for this endpoint?

I added a test for the API endpoint im.history.others. i basically copied the test for im.history, it should work under the same basic principle. Let me know if you need anything else.

**Comment by BlackFenix2** [comment]
I see the Tests didn't like my PR :(. ill look over the endpoint and update the current branch.

**Comment by BlackFenix2** [comment]
@MarcosSpessatto i might have a bit of an issue with this PR,

the value for RocketChat.settings.get('API_Enable_Direct_Message_History_EndPoint') is true inside the 'im.messages.others endpoint but is false on the "im.history.others" endpoint. the tests for both are set up exactly the same so i am not sure why this value is different.

**Comment by MarcosSpessatto** [comment]
@BlackFenix2 sorry for the long delay, I'll take a look on this now.

**Comment by MarcosSpessatto** [comment]
@BlackFenix2 it seems like the endpoint you are creating, already exists, but with another name, [see here](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-api/server/v1/im.js#L234). Of course it is with the wrong name, based on what the setting in the admin panel says. Maybe we can only change the name of the endpoint in the description of the setting, or maybe we can depreciate the old endpoint(`im.messages.others`) and use your endpoint instead. Please let me know what you think.

**Comment by BlackFenix2** [comment]
Wait a sec, I thought the endpoint 'im.messages.others' was the current endpoint. Are we looking to depreciate 'im.history.others'?

I was trying to find a way to search messages within a specific date/time range. Would I be better off coding against im.messages.others? I tried to use query expressions to get messages within a date/time range but I always got null.

**Comment by MarcosSpessatto** [comment]
> Wait a sec, I thought the endpoint 'im.messages.others' was the current endpoint. Are we looking to depreciate 'im.history.others'?

No, I told that we could depreciated `im.messages.others` and use the endpoint you are creating instead.

> I was trying to find a way to search messages within a specific date/time range. Would I be better off coding against im.messages.others?

Yes, I think you can get the messages with this endpoint.

> I tried to use query expressions to get messages within a date/time range but I always got null.

I don't know what server version you are running, because recently (`v0.72`) we introduce the EJSON structure to support these date queries, [PR](https://github.com/RocketChat/Rocket.Chat/pull/12706). [Here you can see a little example searching by dates](https://rocket.chat/docs/developer-guides/rest-api/query-and-fields-info/#query-example).

**Comment by BlackFenix2** [comment]
oh, the query examples for date/time weren't listed last i checked. 

i see my problem, i was using `{"_updatedAt":{"$lt":"2018-12-21T15:27:28.202Z"}}`

rather than `{"_updatedAt":{"$lt":{"$date":"2018-12-21T15:27:28.202Z"}}}`

since the intention of the PR was to sort messages by date-time, ill close out my PR since the functionality is redundant.

**Pull request closed without merging**